src/components/common/Input/Input.tsx

```diff
@@ -0,0 +1,43 @@
+import { ChangeEvent, FC } from 'react';
+
+import * as S from './Input.style';
+
+interface Props {
+  value: string;
+  onChange: (e: ChangeEvent<HTMLInputElement>) => void;
+  placeholder?: string;
+  error?: boolean;
+  errorMessage?: string;
+  maxLength?: number;
+  name?: string;
+  htmlId?: string;
+}
```

Comment:
```suggestion
interface Props extends InputHTMLAttributes<HTMLInputElement> {
  value: string;
  error?: boolean;
  errorMessage?: string;
}
```

기본 props를 상속받아 사용하면 많은 부분을 줄일 수 있을 것 같아요.
이후에도 `disabled`같은 input의 기본 속성을 수정 없이 쓸 수도 있을 것 같습니다.
`Input`에는 아래처럼만 넘겨줘도 되게 변합니다.
```
<S.Input {...props} />
```

추가적으로 `htmlId `값은 결국 `id`값으로 들어가게 되어서, 혼선을 줄 수 있지 않을까 싶습니다!